projects/components/src/file-upload/file-upload.component.ts

```diff
@@ -46,16 +45,8 @@ import { FileUploadState } from './file-display/file-display';
         </div>
         <div class="sub-text">{{ this.subText }}</div>
       </div>
-      <ht-progress-bar *ngIf="this.showProgress" class="bulk-progress-bar" [progress]="this.progress"></ht-progress-bar>
-      <div class="files-section">
-        <ht-file-display
-          *ngFor="let file of this.files; let index = index"
-          [file]="file"
-          [state]="this.uploadState"
-          (deleteClick)="this.deleteFile(index)"
-        >
-        </ht-file-display>
-      </div>
+
+      <ng-content></ng-content>
```

Comment:
So we're moving the logic of the file display and progress to the parent?

Reply:
yes. This makes the api more flexible and also allows the consumer to either show one progress bar for bulk upload or many for individual file upload.